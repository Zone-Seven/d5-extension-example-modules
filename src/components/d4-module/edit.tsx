// External Dependencies.
import React, { ReactElement } from 'react';
import classnames from 'classnames';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';

// Local Dependencies.
import { D4ModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { ModuleStyles } from './styles';
import { moduleClassnames } from './module-classnames';
import { ModuleScriptData } from './module-script-data';

/**
 * Divi 4 Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {D4ModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const D4ModuleEdit = (props: D4ModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
  } = props;

  // Merge module default values with module attributes.
  const moduleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  const HeadingLevel = moduleAttrs?.titleFont?.font?.desktop?.value?.headingLevel;
  const title        = moduleAttrs?.title?.desktop?.value;
  const content      = moduleAttrs?.content?.desktop?.value;

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      classnamesFunction={moduleClassnames}
      scriptDataComponent={ModuleScriptData}
    >
      <HeadingLevel className="d4_module_title">{title}</HeadingLevel>
      <div
        className="d4_module_content"
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </ModuleContainer>
  );
}

export {
  D4ModuleEdit,
};
