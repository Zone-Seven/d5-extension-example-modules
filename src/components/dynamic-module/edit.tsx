// External Dependencies.
import React, { ReactElement, useEffect } from 'react';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';

// Local Dependencies.
import { DynamicModuleEditProps } from './types';
import { ModuleStyles } from './styles';
import { defaultAttrs } from './constants';
import { useGetRecentPosts } from './hooks/get-recent-posts';
import { map } from 'lodash';
import { __ } from '@wordpress/i18n';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {DynamicModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const DynamicModuleEdit = (props: DynamicModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
  } = props;

  // Merge module default values with module attributes.
  const moduleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  const {getPosts, posts, isLoading} = useGetRecentPosts();

  const title         = moduleAttrs?.title?.desktop?.value;
  const numberOfPosts = parseInt(moduleAttrs?.numberOfPosts?.desktop?.value);
  const TitleHeading  = moduleAttrs?.titleFont?.font?.desktop?.value?.headingLevel;
  const PostTitleHeading  = moduleAttrs?.postTitleFont?.font?.desktop?.value?.headingLevel;

  /**
   * Fetches new Portfolio Posts on parameter changes.
   */
   useEffect(() => {
    getPosts(numberOfPosts);
  }, [numberOfPosts]);

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
    >
      {
        ! isLoading && (
          <>
            {
              title && (
                <TitleHeading className="et_pb_dynamic_module_title">{title}</TitleHeading>
              )
            }
            <div className="et_pb_dynamic_module_post_items">
              {
                map(posts, (post) => (
                    <div className="et_pb_dynamic_module_post_item">
                      <PostTitleHeading className="et_pb_dynamic_module_post_item_title">
                        <a href={post?.link} onClick={() => false}>{post?.title?.rendered}</a>
                      </PostTitleHeading>
                      <div className="et_pb_dynamic_module_post_item_content" dangerouslySetInnerHTML={{__html: post?.excerpt?.rendered}} />
                    </div>
                  )
                )
              }
            </div>
          </>
        )
      }
      {
        ! isLoading && posts.length < 1 && (
          <div>{__('No post found.', 'd5-module-extension-example')}</div>
        )
      }
      {
        isLoading && (
          <div>{__('Loading...', 'd5-module-extension-example')}</div>
        )
      }
    </ModuleContainer>
  );
}

export {
  DynamicModuleEdit,
};
