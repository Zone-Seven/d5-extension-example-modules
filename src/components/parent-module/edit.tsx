// External Dependencies.
import React, { ReactElement } from 'react';

// Divi Dependencies.
import { ChildModulesContainer, ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';

// Local Dependencies.
import { ParentModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { ModuleStyles } from './styles';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {ParentModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const ParentModuleEdit = (props: ParentModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
    childrenIds,
  } = props;

  // Merge module default values with module attributes.
  const moduleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      tag="ul"
    >
      <ChildModulesContainer ids={childrenIds} parentId={id} />
    </ModuleContainer>
  );
}

export {
  ParentModuleEdit,
};
