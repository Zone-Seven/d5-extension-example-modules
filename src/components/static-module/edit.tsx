// External Dependencies.
import React, { ReactElement } from 'react';
import classnames from 'classnames';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';

// Local Dependencies.
import { StaticModuleEditProps } from './types';
import { ModuleStyles } from './styles';
import { moduleClassnames } from './module-classnames';
import { ModuleScriptData } from './module-script-data';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {StaticModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const StaticModuleEdit = (props: StaticModuleEditProps): ReactElement => {
  const {
    attrs,
    elements,
    id,
    name,
  } = props;

  // Get module attributes.
  const imageSrc                = attrs?.image?.innerContent?.desktop?.value?.src;
  const imageAlt                = attrs?.image?.innerContent?.desktop?.value?.alt;

  return (
    <ModuleContainer
      attrs={attrs}
      elements={elements}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      classnamesFunction={moduleClassnames}
      scriptDataComponent={ModuleScriptData}
    >
      <div className="static-module__image">
        <img src={imageSrc} alt={imageAlt} />
      </div>
      <div className="static-module__content-container">
        {elements.render({
          attrName: 'title',
        })}
        <div className="static-module__content">
          {elements.render({
            attrName: 'content',
          })}
        </div>
      </div>
    </ModuleContainer>
  );
}

export {
  StaticModuleEdit,
};
