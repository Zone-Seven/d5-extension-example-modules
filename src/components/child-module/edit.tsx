// External Dependencies.
import React, { ReactElement } from 'react';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';
import { processFontIcon } from '@divi/icon-library';

// Local Dependencies.
import { ChildModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { defaultAttrs as parentDefaultAttrs } from '../parent-module/constants';
import { ModuleStyles } from './styles';
import { moduleClassnames } from './module-classnames';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {ChildModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const ChildModuleEdit = (props: ChildModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
    parentAttrs,
  } = props;

  // Merge parent module default values with parent module attributes.
  const parentModuleAttrs = mergeAttrs({
    defaultAttrs: parentDefaultAttrs,
    attrs: parentAttrs,
  });

  // Merge module default values with module attributes.
  const childModuleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  const moduleAttrs = {
    ...childModuleAttrs,
    icon: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.icon,
      attrs: childModuleAttrs?.icon,
    }),
    iconColor: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.iconColor,
      attrs: childModuleAttrs?.iconColor,
    }),
    iconSize: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.iconSize,
      attrs: childModuleAttrs?.iconSize,
    }),
    titleFont: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.titleFont,
      attrs: childModuleAttrs?.titleFont,
    }),
    text: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.text,
      attrs: childModuleAttrs?.text,
    }),
    bodyFont: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.bodyFont,
      attrs: childModuleAttrs?.bodyFont,
    }),
  };
  
  const icon    = moduleAttrs?.icon?.desktop?.value;
  const title   = moduleAttrs?.title?.desktop?.value;
  const content = moduleAttrs?.content?.desktop?.value;

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      classnamesFunction={moduleClassnames}
      tag="li"
    >
      { icon && (
        <div className="child-module__icon et-pb-icon">
            {processFontIcon(icon)}
        </div>
      )}
      <div className="child-module__content-container">
        <div className="child-module__title">{title}</div>
        <div
          className="child-module__content"
          dangerouslySetInnerHTML={{ __html: content }}
        />
      </div>
    </ModuleContainer>
  );
}

export {
  ChildModuleEdit,
};
