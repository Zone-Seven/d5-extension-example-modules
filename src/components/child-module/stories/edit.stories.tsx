import {
  createElement,
  ReactElement,
} from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { Story } from '@storybook/react';
import {
  defaultAttributes,
  customIcon,
  iconStyle,
  titleStyle,
  contentStyle,
} from '../__mock-data__/attrs';

import { ChildModuleEdit } from '../edit';
import { ChildModuleEditProps } from '../types';
import { select } from '@divi/data';
import { childModule } from '..';
import { isNull, isUndefined, omit } from 'lodash';
import { registerModule, registerModuleLibraryStore } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerModals } from '@divi/modal-library';
import { registerAppUiStore } from '@divi/app-ui';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { registerEditPostStore } from '@divi/edit-post';
import { registerSettingsStore } from '@divi/settings';
import { registerHistoryStore } from '@divi/history';
import { registerAppPreferencesStore } from '@divi/app-preferences';

// Create template to render argument given.
const templateDividerEdit: Story<ChildModuleEditProps> = args => createElement(ChildModuleEdit, args);

const name = 'example/child-module';

// Stories.
const DefaultValues      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
DefaultValues.args       = {
  name,
  id:    'child-module-1',
  attrs: defaultAttributes,
};
DefaultValues.parameters = {
  docs: {
    description: {
      story: 'Default attributes that gets added when Child module is added.',
    },
  },
};

const CustomIcon      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
CustomIcon.args       = {
  name,
  id:    'child-module-2',
  attrs: customIcon,
};
CustomIcon.parameters = {
  docs: {
    description: {
      story: 'Display selected icon properly when the icon value is set.',
    },
  },
};

const IconStyle      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
IconStyle.args       = {
  name,
  id:    'child-module-3',
  attrs: iconStyle,
};
IconStyle.parameters = {
  docs: {
    description: {
      story: 'Display icon style properly when the icon style is set.',
    },
  },
};

const TitleStyle      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
TitleStyle.args       = {
  name,
  id:    'child-module-4',
  attrs: titleStyle,
};
TitleStyle.parameters = {
  docs: {
    description: {
      story: 'Display title style properly when the title style is set.',
    },
  },
};

const ContentStyle      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
ContentStyle.args       = {
  name,
  id:    'child-module-5',
  attrs: contentStyle,
};
ContentStyle.parameters = {
  docs: {
    description: {
      story: 'Display content style properly when the content style is set.',
    },
  },
};



export {
  DefaultValues,
  CustomIcon,
  IconStyle,
  TitleStyle,
  ContentStyle,
};

export default {
  title: 'Modules/Child Module/Edit',
  decorators: [
    ((story: () => ReactElement): ReactElement => {
      registerAppPreferencesStore();
      registerAppUiStore();
      registerEventsStore();
      registerKeyboardShortcutsStore();
      registerModals();
      registerSettingsStore();
      registerSerializedPostStore();
      registerEditPostStore();
      registerHistoryStore();
      if (isNull(select('divi/module-library'))) {
        registerModuleLibraryStore();
      }
      const theChildModule  = select('divi/module-library').getModule(childModule?.metadata?.name);

      if (isUndefined(theChildModule)) {
        registerModule(childModule.metadata, omit(childModule, 'metadata'));
      }
      return story();
    }),
  ],
};
