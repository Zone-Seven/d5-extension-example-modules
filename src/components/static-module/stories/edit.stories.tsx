import {
  createElement,
  ReactElement,
} from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { Story } from '@storybook/react';
import {
  defaultAttributes,
  contentStyle,
  customBackground,
  customImage,
  imageStyle,
  titleStyle,
} from '../__mock-data__/attrs';

import { StaticModuleEdit } from '../edit';
import { StaticModuleEditProps } from '../types';
import { staticModule } from '..';
import { registerModule } from '@divi/module-library';
import { omit } from 'lodash';

// Create template to render argument given.
const templateDividerEdit: Story<StaticModuleEditProps> = args => createElement(StaticModuleEdit, args);

const name = 'divi/static-module';

// Stories.
const DefaultValues      = templateDividerEdit.bind({}) as Story<StaticModuleEditProps>;
DefaultValues.args       = {
  name,
  id:    'static-module-1',
  attrs: defaultAttributes,
};
DefaultValues.parameters = {
  docs: {
    description: {
      story: 'Default attributes that gets added when Static module is added.',
    },
  },
};

const ContentStyle      = templateDividerEdit.bind({}) as Story<StaticModuleEditProps>;
ContentStyle.args       = {
  name,
  id:    'static-module-2',
  attrs: contentStyle,
};
ContentStyle.parameters = {
  docs: {
    description: {
      story: 'Display content text style properly when the content text style is set.',
    },
  },
};

const CustomBackground      = templateDividerEdit.bind({}) as Story<StaticModuleEditProps>;
CustomBackground.args       = {
  name,
  id:    'static-module-3',
  attrs: customBackground,
};
CustomBackground.parameters = {
  docs: {
    description: {
      story: 'Display custom background properly when the custom background is set.',
    },
  },
};

const CustomImage      = templateDividerEdit.bind({}) as Story<StaticModuleEditProps>;
CustomImage.args       = {
  name,
  id:    'static-module-4',
  attrs: customImage,
};
CustomImage.parameters = {
  docs: {
    description: {
      story: 'Display custom image properly when the custom image is set.',
    },
  },
};

const ImageStyle      = templateDividerEdit.bind({}) as Story<StaticModuleEditProps>;
ImageStyle.args       = {
  name,
  id:    'static-module-5',
  attrs: imageStyle,
};
ImageStyle.parameters = {
  docs: {
    description: {
      story: 'Display image style properly when the image style is set.',
    },
  },
};

const TitleStyle      = templateDividerEdit.bind({}) as Story<StaticModuleEditProps>;
TitleStyle.args       = {
  name,
  id:    'static-module-6',
  attrs: titleStyle,
};
TitleStyle.parameters = {
  docs: {
    description: {
      story: 'Display title text style properly when the title text style is set.',
    },
  },
};


export {
  DefaultValues,
  TitleStyle,
  ContentStyle,
  CustomBackground,
  CustomImage,
  ImageStyle,
};

export default {
  title: 'Modules/Static Module/Edit',
  decorators: [
    ((story: () => ReactElement): ReactElement => {
      registerModule(staticModule.metadata, omit(staticModule, 'metadata'));
      return story();
    }),
  ],
};
