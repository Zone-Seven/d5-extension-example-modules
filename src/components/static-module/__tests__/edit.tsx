import { createElement } from 'react';
import { shallow } from 'enzyme';
import {
  forEach,
  pickBy,
  has,
  get,
} from 'lodash';

import { registerAppUiStore } from '@divi/app-ui';
import { registerModuleLibraryStore } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerModals } from '@divi/modal-library';
import { registerEditPostStore } from '@divi/edit-post';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { StaticModuleEdit } from '../edit';
import { dispatch } from '@divi/data';
import { staticModule } from '..';

import * as editStories  from '../stories/edit.stories';

// Only test story that has description. This naturally excludes `default` export from being included as test.
const testableStories = pickBy(editStories, story => has(story, 'parameters.docs.description.story')) as Omit<typeof editStories, 'default'>;


beforeAll(() => {
  registerModuleLibraryStore();
  dispatch('divi/module-library').addModule(staticModule);
});

beforeEach(() => {
  registerAppUiStore();
  registerEventsStore();
  registerKeyboardShortcutsStore();
  registerModals();
  registerSerializedPostStore();
  registerEditPostStore();
});

describe('<StaticModuleEdit />', () => {
  // Snapshot test based on storybook configuration.
  // Prevents regression due to unintentionally modifying component.
  forEach(testableStories, testableStory => {
    it(`Should match the snapshot when "${get(testableStory, 'parameters.docs.description.story') as string}"`, () => {
      expect(
        shallow(createElement(StaticModuleEdit, testableStory.args)).html(),
      ).toMatchSnapshot();
    });
  });
});
