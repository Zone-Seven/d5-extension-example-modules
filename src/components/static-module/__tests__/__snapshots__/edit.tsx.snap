// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticModuleEdit /> Should match the snapshot when "Default attributes that gets added when Static module is added." 1`] = `"<div class=\\"et_pb_static_module et_pb_text_align_undefined et_pb_bg_layout_light et_pb_module\\" data-id=\\"static-module-1\\"><div class=\\"et_pb_static_module_image\\"><img src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTA4MCIgaGVpZ2h0PSI1NDAiIHZpZXdCb3g9IjAgMCAxMDgwIDU0MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPHBhdGggZmlsbD0iI0VCRUJFQiIgZD0iTTAgMGgxMDgwdjU0MEgweiIvPgogICAgICAgIDxwYXRoIGQ9Ik00NDUuNjQ5IDU0MGgtOTguOTk1TDE0NC42NDkgMzM3Ljk5NSAwIDQ4Mi42NDR2LTk4Ljk5NWwxMTYuMzY1LTExNi4zNjVjMTUuNjItMTUuNjIgNDAuOTQ3LTE1LjYyIDU2LjU2OCAwTDQ0NS42NSA1NDB6IiBmaWxsLW9wYWNpdHk9Ii4xIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgICAgICA8Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjA1IiBmaWxsPSIjMDAwIiBjeD0iMzMxIiBjeT0iMTQ4IiByPSI3MCIvPgogICAgICAgIDxwYXRoIGQ9Ik0xMDgwIDM3OXYxMTMuMTM3TDcyOC4xNjIgMTQwLjMgMzI4LjQ2MiA1NDBIMjE1LjMyNEw2OTkuODc4IDU1LjQ0NmMxNS42Mi0xNS42MiA0MC45NDgtMTUuNjIgNTYuNTY4IDBMMTA4MCAzNzl6IiBmaWxsLW9wYWNpdHk9Ii4yIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgIDwvZz4KPC9zdmc+Cg==\\"/></div><div class=\\"et_pb_static_module_content_container\\"><h2 class=\\"et_pb_static_module_title\\">Your Title Goes Here</h2><div class=\\"et_pb_static_module_content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></div>"`;

exports[`<StaticModuleEdit /> Should match the snapshot when "Display content text style properly when the content text style is set." 1`] = `"<div class=\\"et_pb_static_module et_pb_text_align_undefined et_pb_bg_layout_light et_pb_module\\" data-id=\\"static-module-2\\"><div class=\\"et_pb_static_module_image\\"><img src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTA4MCIgaGVpZ2h0PSI1NDAiIHZpZXdCb3g9IjAgMCAxMDgwIDU0MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPHBhdGggZmlsbD0iI0VCRUJFQiIgZD0iTTAgMGgxMDgwdjU0MEgweiIvPgogICAgICAgIDxwYXRoIGQ9Ik00NDUuNjQ5IDU0MGgtOTguOTk1TDE0NC42NDkgMzM3Ljk5NSAwIDQ4Mi42NDR2LTk4Ljk5NWwxMTYuMzY1LTExNi4zNjVjMTUuNjItMTUuNjIgNDAuOTQ3LTE1LjYyIDU2LjU2OCAwTDQ0NS42NSA1NDB6IiBmaWxsLW9wYWNpdHk9Ii4xIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgICAgICA8Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjA1IiBmaWxsPSIjMDAwIiBjeD0iMzMxIiBjeT0iMTQ4IiByPSI3MCIvPgogICAgICAgIDxwYXRoIGQ9Ik0xMDgwIDM3OXYxMTMuMTM3TDcyOC4xNjIgMTQwLjMgMzI4LjQ2MiA1NDBIMjE1LjMyNEw2OTkuODc4IDU1LjQ0NmMxNS42Mi0xNS42MiA0MC45NDgtMTUuNjIgNTYuNTY4IDBMMTA4MCAzNzl6IiBmaWxsLW9wYWNpdHk9Ii4yIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgIDwvZz4KPC9zdmc+Cg==\\"/></div><div class=\\"et_pb_static_module_content_container\\"><h2 class=\\"et_pb_static_module_title\\">Your Title Goes Here</h2><div class=\\"et_pb_static_module_content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.et_pb_static_module[data-id=&quot;static-module-2&quot;] .et_pb_static_module_content {font-style: italic; letter-spacing: 1px}.et_pb_static_module[data-id=&quot;static-module-2&quot;] .et_pb_static_module_content {text-align: center}</style></div>"`;

exports[`<StaticModuleEdit /> Should match the snapshot when "Display custom background properly when the custom background is set." 1`] = `"<div class=\\"et_pb_static_module et_pb_text_align_undefined et_pb_bg_layout_light et_pb_module\\" data-id=\\"static-module-3\\"><div class=\\"et_pb_static_module_image\\"><img src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTA4MCIgaGVpZ2h0PSI1NDAiIHZpZXdCb3g9IjAgMCAxMDgwIDU0MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPHBhdGggZmlsbD0iI0VCRUJFQiIgZD0iTTAgMGgxMDgwdjU0MEgweiIvPgogICAgICAgIDxwYXRoIGQ9Ik00NDUuNjQ5IDU0MGgtOTguOTk1TDE0NC42NDkgMzM3Ljk5NSAwIDQ4Mi42NDR2LTk4Ljk5NWwxMTYuMzY1LTExNi4zNjVjMTUuNjItMTUuNjIgNDAuOTQ3LTE1LjYyIDU2LjU2OCAwTDQ0NS42NSA1NDB6IiBmaWxsLW9wYWNpdHk9Ii4xIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgICAgICA8Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjA1IiBmaWxsPSIjMDAwIiBjeD0iMzMxIiBjeT0iMTQ4IiByPSI3MCIvPgogICAgICAgIDxwYXRoIGQ9Ik0xMDgwIDM3OXYxMTMuMTM3TDcyOC4xNjIgMTQwLjMgMzI4LjQ2MiA1NDBIMjE1LjMyNEw2OTkuODc4IDU1LjQ0NmMxNS42Mi0xNS42MiA0MC45NDgtMTUuNjIgNTYuNTY4IDBMMTA4MCAzNzl6IiBmaWxsLW9wYWNpdHk9Ii4yIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgIDwvZz4KPC9zdmc+Cg==\\"/></div><div class=\\"et_pb_static_module_content_container\\"><h2 class=\\"et_pb_static_module_title\\">Your Title Goes Here</h2><div class=\\"et_pb_static_module_content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.et_pb_static_module[data-id=&quot;static-module-3&quot;] {background-color: #7cda24}</style></div>"`;

exports[`<StaticModuleEdit /> Should match the snapshot when "Display custom image properly when the custom image is set." 1`] = `"<div class=\\"et_pb_static_module et_pb_text_align_undefined et_pb_bg_layout_light et_pb_module\\" data-id=\\"static-module-4\\"><div class=\\"et_pb_static_module_image\\"><img src=\\"https://www.elegantthemes.com/images/home/category-thumb-store-large.jpg\\" alt=\\"category thumb store large\\"/></div><div class=\\"et_pb_static_module_content_container\\"><h2 class=\\"et_pb_static_module_title\\">Your Title Goes Here</h2><div class=\\"et_pb_static_module_content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></div>"`;

exports[`<StaticModuleEdit /> Should match the snapshot when "Display image style properly when the image style is set." 1`] = `"<div class=\\"et_pb_static_module et_pb_text_align_undefined et_pb_bg_layout_light et_pb_module\\" data-id=\\"static-module-5\\"><div class=\\"et_pb_static_module_image\\"><img src=\\"https://www.elegantthemes.com/images/home/category-thumb-store-large.jpg\\" alt=\\"category thumb store large\\"/></div><div class=\\"et_pb_static_module_content_container\\"><h2 class=\\"et_pb_static_module_title\\">Your Title Goes Here</h2><div class=\\"et_pb_static_module_content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.et_pb_static_module[data-id=&quot;static-module-5&quot;] .et_pb_static_module_image img {border-bottom-left-radius: 14px !important; border-top-left-radius: 14px !important; border-bottom-right-radius: 14px !important; border-top-right-radius: 14px !important; border-width: 10px !important; border-color: #8300e9 !important; border-style: solid !important}.et_pb_static_module[data-id=&quot;static-module-5&quot;] .et_pb_static_module_image img {box-shadow: 6px 6px 18px 0px rgba(0,0,0,0.3);}</style></div>"`;

exports[`<StaticModuleEdit /> Should match the snapshot when "Display title text style properly when the title text style is set." 1`] = `"<div class=\\"et_pb_static_module et_pb_text_align_undefined et_pb_bg_layout_light et_pb_module\\" data-id=\\"static-module-6\\"><div class=\\"et_pb_static_module_image\\"><img src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTA4MCIgaGVpZ2h0PSI1NDAiIHZpZXdCb3g9IjAgMCAxMDgwIDU0MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPHBhdGggZmlsbD0iI0VCRUJFQiIgZD0iTTAgMGgxMDgwdjU0MEgweiIvPgogICAgICAgIDxwYXRoIGQ9Ik00NDUuNjQ5IDU0MGgtOTguOTk1TDE0NC42NDkgMzM3Ljk5NSAwIDQ4Mi42NDR2LTk4Ljk5NWwxMTYuMzY1LTExNi4zNjVjMTUuNjItMTUuNjIgNDAuOTQ3LTE1LjYyIDU2LjU2OCAwTDQ0NS42NSA1NDB6IiBmaWxsLW9wYWNpdHk9Ii4xIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgICAgICA8Y2lyY2xlIGZpbGwtb3BhY2l0eT0iLjA1IiBmaWxsPSIjMDAwIiBjeD0iMzMxIiBjeT0iMTQ4IiByPSI3MCIvPgogICAgICAgIDxwYXRoIGQ9Ik0xMDgwIDM3OXYxMTMuMTM3TDcyOC4xNjIgMTQwLjMgMzI4LjQ2MiA1NDBIMjE1LjMyNEw2OTkuODc4IDU1LjQ0NmMxNS42Mi0xNS42MiA0MC45NDgtMTUuNjIgNTYuNTY4IDBMMTA4MCAzNzl6IiBmaWxsLW9wYWNpdHk9Ii4yIiBmaWxsPSIjMDAwIiBmaWxsLXJ1bGU9Im5vbnplcm8iLz4KICAgIDwvZz4KPC9zdmc+Cg==\\"/></div><div class=\\"et_pb_static_module_content_container\\"><h1 class=\\"et_pb_static_module_title\\">Your Title Goes Here</h1><div class=\\"et_pb_static_module_content\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div><style class=\\"et-vb-module-styles\\">.et_pb_static_module[data-id=&quot;static-module-6&quot;] .et_pb_static_module_title {font-weight: 300; text-transform: uppercase; color: #8300e9; font-size: 36px}</style></div>"`;
